lib/util/git/url.ts

```diff
@@ -4,18 +4,31 @@ import * as hostRules from '../host-rules';
 
 export function getHttpUrl(url: string, token?: string): string {
   const parsedUrl = GitUrlParse(url);
+
   parsedUrl.token = token;
-  return parsedUrl.toString('https');
+
+  const protocol = /^https?$/.exec(parsedUrl.protocol)
+    ? parsedUrl.protocol
+    : 'https';
+  return parsedUrl.toString(protocol);
```

Comment:
So this will allow `http://` protocol now?

Reply:
yes correct, I guess this could be considered the most controversial change here..
I personally know and think that configuring TLS is cheap and easy but for legacy reasons I think the use case I have will never see TLS configured. And I'm not sure if this should really be completely forbidden for now